### ISSUE TITLE
Getting error with Content Security Policy (CSP) because tags have inline-scripts

### ISSUE THREAD

**Issue by rodolvbg-divineslns** [makinacorpus/django-leaflet#334]
Hello,

I'm using Content Security Policy and I get this error when use leaflet tags

![image](https://user-images.githubusercontent.com/75988609/122423584-01daf600-cf5c-11eb-920a-a3bdaa4fc346.png)

I see that `{% leaflet_js %}`, `{% leaflet_css %}` and `{% leaflet_map %}` tags contain unsafe inline code. Maybe it's necessary insert into an external script

**Comment by Gagaro** [comment]
Hi,

JS/CSS are inlines because it needs a lot of context data, it was the most convenient way to do it.

https://github.com/makinacorpus/django-leaflet/blob/master/leaflet/templates/leaflet/_leaflet_map.html
https://github.com/makinacorpus/django-leaflet/blob/master/leaflet/templates/leaflet/css.html
https://github.com/makinacorpus/django-leaflet/blob/master/leaflet/templates/leaflet/js.html

You can overwrite those templates in your own app template folder if you app is before django-leaflet in `INSTALLED_APPS`. This allows you to add what you need to make it work (unsafe-inline, hash, ...).

If you have a way to make it work for everyone, do not hesitate to do a PR :+1: .

**Comment by rodolvbg-divineslns** [comment]
> Hi,
> 
> JS/CSS are inlines because it needs a lot of context data, it was the most convenient way to do it.
> 
> https://github.com/makinacorpus/django-leaflet/blob/master/leaflet/templates/leaflet/_leaflet_map.html
> https://github.com/makinacorpus/django-leaflet/blob/master/leaflet/templates/leaflet/css.html
> https://github.com/makinacorpus/django-leaflet/blob/master/leaflet/templates/leaflet/js.html
> 
> You can overwrite those templates in your own app template folder if you app is before django-leaflet in `INSTALLED_APPS`. This allows you to add what you need to make it work (unsafe-inline, hash, ...).
> 
> If you have a way to make it work for everyone, do not hesitate to do a PR  .

Sorry, could you document me on how to do a PR? I have created the branch on my device and it won't let me upload it

**Comment by Gagaro** [comment]
You first need to fork the repository (button on the upper right), then you push your branch in your fork:

``` bash
git remote add fork git@github.com:rodolvbg-divineslns/django-leaflet.git
git push fork
```

Then you can start a new PR and click on `compare across forks` and choosing your fork and branch.

**Comment by Gagaro** [comment]
Fixed by #371
